src/ua/dymohlo/parametersvaliditycheck/Main.java

```diff
@@ -0,0 +1,89 @@
+        double side = 0;
+        Scanner scan = new Scanner(System.in);
+        System.out.println("Enter the length of the side of the square: ");
+        while (side <= 0) {
+            try {
+                side = scan.nextDouble();
+                if (side <= 0) {
+                    throw new ValidValues("Such a square does not exist!");
+                }
+                Figure square = new Square(side);
+                System.out.println("Perimeter of a square is " + (square.perimeter()));
+                System.out.println("Area of a square is " + (square.area()));
+            } catch (ValidValues e) {
+                e.printStackTrace();
+                System.out.println("Enter another value");
+            }
+        }
+    }
+
+    private static void circleParametersValidityCheck() {
+        double rad = 0;
+        Scanner scan = new Scanner(System.in);
+        System.out.println("Enter the radius of circle:");
+        while (rad <= 0) {
+            try {
+                rad = scan.nextDouble();
+                if (rad <= 0) {
+                    throw new ValidValues("The radius of the circle must be positive!");
+                } else {
+                    Figure circle = new Circle(rad);
+                    System.out.println("Perimeter of a circle is " + (circle.perimeter()));
+                    System.out.println("Area of a circle is " + (circle.area()));
+                }
+            } catch (ValidValues e) {
+                e.printStackTrace();
+                System.out.println("Enter another value");
+            }
+        }
+    }
+
+    private static void triangleParametersValidityCheck() {
+        Scanner scan = new Scanner(System.in);
+        double sideA = inputValue(scan, "Enter the length of side A:", "Such a triangle does not exist!");
+        double sideB = inputValue(scan, "Enter the length of side B:", "Such a triangle does not exist!");
+        double sideC = inputValue(scan, "Enter the length of side C:", "Such a triangle does not exist!");
+        double sin = inputValue(scan, "Enter the sine of the angle between the sides A and B: ", "The sine of the angle cannot be zero or have a negative value!");
+        double height = inputValue(scan, "Enter the height of the triangle: ", "The height of the triangle must be greater than zero!");
+
+        Triangle triangle = new Triangle(sideA, sideB, sideC);
+        Triangle triangle2 = new Triangle(sideA, sideB, sin);
+        Triangle triangle3 = new Triangle(sideA, height);
+        System.out.println("Perimeter of a triangle is " + (triangle.perimeter()));
+        System.out.println("Area of a triangle found by Heron's formula is " + (triangle.area()));
+        System.out.println("Area of a triangle found by the product of two sides and the angle between them is " + (triangle2.area(sideA, sideB, sin)));
+        System.out.println("Area of a triangle found by a base height product is " + (triangle3.area(sideA, height)));
```

Comment:
triangle3 has it's own state with the sideA, you do not need to pass sideA's length to this method